src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -47,6 +47,22 @@
 logger = logging.getLogger("pyfluent.launcher")
 
 
+def _remove_unused_args(fluent_launch_mode: LaunchMode = None, **kwargs):
+
+    def _remove_keys(unused_args, **kargs):
+        for arg in unused_args:
+            kargs.pop(arg, None)
+        return kargs
+
+    if fluent_launch_mode == LaunchMode.STANDALONE or LaunchMode.PIM:
+        return _remove_keys(
+            ["start_container", "container_dict", "dry_run", "scheduler_options"],
+            **kwargs,
+        )
+    elif fluent_launch_mode == LaunchMode.CONTAINER:
+        return _remove_keys(["start_container", "scheduler_options"], **kwargs)
```

Comment:
Given that the individual constructors specify what is allowed, this is actually not needed.

Reply:
@seanpearsonuk  Yes, that's right, but these arguments are coming from `launch_fluent` not from the individual constructor.

Reply:
We have removed this code now. Thank you.